.github/workflows/kind-e2e.yaml

```diff
@@ -0,0 +1,197 @@
+name: KinD e2e tests
+
+on:
+  push:
+    branches: [ 'master' ]
```

Comment:
I think we should drop this. See https://github.com/knative-sandbox/.github/pull/18 for reference.

Reply:
I don't mind it, why drop it? it could be signal in the future

Reply:
A signal of which kind? This should effectively just run the tests twice, while the second time nobody will check.

Not hardwired here at all, but it seems wasteful.

Reply:
My rule of PRs is to punt on anything contentious.  Let's kill it and add it back when we need it.